[ci skip] CI: Add ALCF GitLab test scripts

masterleinad opened the pull request:
These are the scripts used for https://gitlab-ci.alcf.anl.gov/anl/kokkos_math/kokkos_core and https://gitlab-sunspot.alcf.anl.gov/anl/CSC250STPM18_CNDA/kokkos right now.

dalg24 commented on .gitlab/sunspot-gitlab-ci.yml:
```diff
@@ -0,0 +1,40 @@
+        -D OUTPUT_JUNIT_FILE=result_pvc1550.xml
+        -S CTestRun.cmake
+        -D CTEST_SITE="gitlab-sunspot.alcf.anl.gov"
+        -D CTEST_BUILD_NAME="${CTEST_BUILD_NAME}"
```
Do we need both the environment variable and that variable set here?

masterleinad replied:
No, we don't need the environment variable.

dalg24 replied:
I would recommend getting rid of it then

dalg24 reviewed (approved):
Are these configuration files going to be used directly in these GitLab instances and if the answer is no, what is the plan for keeping them in sync?

dalg24 commented:
> Are these configuration files going to be used directly in these GitLab instances and if the answer is no, what is the plan for keeping them in sync?

Please address this comment before I merge

masterleinad commented:
> Are these configuration files going to be used directly in these GitLab instances and if the answer is no, what is the plan for keeping them in sync?

If we really want to enforce controlling it from here, we should just do repository mirroring.

masterleinad commented:
> If we really want to enforce controlling it from here, we should just do repository mirroring.

I added appropriate changes here and switched the GitLab instances over.
